### PR TITLE
[ROIAlign-9] Extend nGraph Python API for operation "ROIAlign-9"

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - [x] Extend nGraph Python API for operation "ROIAlign-9"
 - [x] add tests

### Tickets:
 - 82327

### CODE REVIEW

File: src/bindings/python/src/compatibility/ngraph/opset9/ops.py [hunk 1]
```diff
@@ -48,6 +49,48 @@ def eye(
...
+    :param batch_indices: Tensor with each element denoting the index of
+                          the corresponding image in the batch.
+    :param pooled_h: Height of the ROI output feature map.
+    :param pooled_w: Width of the ROI output feature map.
+    :param sampling_ratio: Number of bins over height and width to use to calculate
+                           each output feature map element.
+    :param spatial_scale: Multiplicative spatial scale factor to translate ROI coordinates.
+    :param mode: Method to perform pooling to produce output feature map elements.
+    :param aligned_mode: specifies how to transform the coordinate in original tensor to the resized tensor.
+                         Mode 'asymmetric' is default.
```
Comment (jiwaszki) [inline src/bindings/python/src/compatibility/ngraph/opset9/ops.py:77]:
```suggestion
    :param mode: Method to perform pooling to produce output feature map elements.
    :param aligned_mode: Specifies how to transform the coordinate in original tensor to the resized tensor.
                         Mode 'asymmetric' is the default value. Optional.
```

Comment (jiwaszki) [reply src/bindings/python/src/compatibility/ngraph/opset9/ops.py:77]: You may also specify what values of `mode`/`aligned_mode` are possible.

File: src/bindings/python/src/openvino/runtime/opset9/ops.py [hunk 2]
```diff
@@ -49,6 +49,48 @@ def eye(
...
+    :param batch_indices: Tensor with each element denoting the index of
+                          the corresponding image in the batch.
+    :param pooled_h: Height of the ROI output feature map.
+    :param pooled_w: Width of the ROI output feature map.
+    :param sampling_ratio: Number of bins over height and width to use to calculate
+                           each output feature map element.
+    :param spatial_scale: Multiplicative spatial scale factor to translate ROI coordinates.
+    :param mode: Method to perform pooling to produce output feature map elements.
+    :param aligned_mode: specifies how to transform the coordinate in original tensor to the resized tensor.
+                         Mode 'asymmetric' is default.
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/opset9/ops.py:77]: Same as for previous comments.

File: src/bindings/python/tests/test_ngraph/test_reduction.py [hunk 3]
```diff
@@ -149,6 +149,38 @@ def test_roi_align():
...
+    spatial_scale = np.float32(1)
+
+    node = ov.roi_align(
+        data_parameter,
+        rois_parameter,
+        batch_indices_parameter,
+        pooled_h,
+        pooled_w,
+        sampling_ratio,
+        np.float32(spatial_scale),
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_ngraph/test_reduction.py:174]: Is there a reason to create "double scalar" here? It is already declared as `np.float32` above.

Could you add a different test when using simple float value without numpy?

**Review by jiwaszki — COMMENTED** [review]
Only small nits about docstrings and tests. Overall it is OK.

File: src/bindings/python/tests/test_ngraph/test_reduction.py [hunk 4]
```diff
@@ -149,6 +149,38 @@ def test_roi_align():
...
 
+@pytest.mark.parametrize(
+    ("data_shape", "rois", "batch_indices", "pooled_h", "pooled_w", "sampling_ratio", "spatial_scale", "mode", "aligned_mode", "expected_shape"),
+    [
+        ([2, 3, 5, 6], [7, 4], [7], 2, 2, 1, 1.0, "avg", "asymmetric", [7, 3, 2, 2]),
+        ([10, 3, 5, 5], [7, 4], [7], 3, 4, 1, 1.0, "avg", "half_pixel_for_nn", [7, 3, 3, 4]),
+        ([10, 3, 5, 5], [3, 4], [3], 3, 4, 1, 1.0, "avg", "half_pixel", [3, 3, 3, 4]),
+    ],
+)
+def test_roi_alignv9(data_shape, rois, batch_indices, pooled_h, pooled_w, sampling_ratio, spatial_scale, mode, aligned_mode, expected_shape):
```
Comment (mitruska) [inline src/bindings/python/tests/test_ngraph/test_reduction.py:160]: I'm not sure this ROIAlign test really fit to the `test_reduction.py` file, I would suggest to move it to the `test_create_op.py`.
(Same for /compatibility)

File: src/bindings/python/tests/test_ngraph/test_reduction.py [hunk 5]
```diff
@@ -149,6 +149,38 @@ def test_roi_align():
...
+        sampling_ratio,
+        np.float32(spatial_scale),
+        mode,
+        aligned_mode,
+    )
+
+    assert node.get_type_name() == "ROIAlign"
+    assert node.get_output_size() == 1
+    assert list(node.get_output_shape(0)) == expected_shape
+
```
Comment (mitruska) [inline src/bindings/python/tests/test_ngraph/test_reduction.py:182]: Please add assert for `node.get_output_element_type(0)`
(Same for /compatibility)

File: src/bindings/python/tests/test_ngraph/test_reduction.py (hunk 3 above)
Comment (bszmelcz) [reply src/bindings/python/tests/test_ngraph/test_reduction.py:174]: Sure :)

File: src/bindings/python/src/compatibility/ngraph/opset9/ops.py (hunk 1 above)
Comment (bszmelcz) [reply src/bindings/python/src/compatibility/ngraph/opset9/ops.py:77]: great idea!

File: src/bindings/python/tests_compatibility/test_ngraph/test_create_op.py [hunk 6]
```diff
@@ -249,6 +249,39 @@ def test_deformable_psroi_pooling(dtype):
...
+        ([10, 3, 5, 5], [7, 4], [7], 3, 4, 1, 1.0, "avg", "half_pixel_for_nn", [7, 3, 3, 4]),
+        ([10, 3, 5, 5], [3, 4], [3], 3, 4, 1, 1.0, "avg", "half_pixel", [3, 3, 3, 4]),
+    ],
+)
+def test_roi_align(data_shape, rois, batch_indices, pooled_h, pooled_w, sampling_ratio, spatial_scale, mode, aligned_mode, expected_shape):
+    data_parameter = ng.parameter(data_shape, name="Data", dtype=np.float32)
+    rois_parameter = ng.parameter(rois, name="Rois", dtype=np.float32)
+    batch_indices_parameter = ng.parameter(batch_indices, name="Batch_indices", dtype=np.int32)
+
+    spatial_scale = np.float32(1)
```
Comment (jiwaszki) [inline src/bindings/python/tests_compatibility/test_ngraph/test_create_op.py:265]:
```suggestion
@pytest.mark.parametrize(
    ("data_shape", "rois", "batch_indices", "pooled_h", "pooled_w", "sampling_ratio", "spatial_scale", "mode", "aligned_mode", "expected_shape"),
    [
        ([2, 3, 5, 6], [7, 4], [7], 2, 2, 1, 1.0, "avg", "asymmetric", [7, 3, 2, 2]),
        ([10, 3, 5, 5], [7, 4], [7], 3, 4, 1, 1.0, "avg", "half_pixel_for_nn", [7, 3, 3, 4]),
        ([10, 3, 5, 5], [3, 4], [3], 3, 4, 1, 1.0, "avg", "half_pixel", [3, 3, 3, 4]),
        ([10, 3, 5, 5], [3, 4], [3], 3, 4, 1, numpy.float(1), "avg", "half_pixel", [3, 3, 3, 4]),
    ],
)
def test_roi_align(data_shape, rois, batch_indices, pooled_h, pooled_w, sampling_ratio, spatial_scale, mode, aligned_mode, expected_shape):
    data_parameter = ng.parameter(data_shape, name="Data", dtype=np.float32)
    rois_parameter = ng.parameter(rois, name="Rois", dtype=np.float32)
    batch_indices_parameter = ng.parameter(batch_indices, name="Batch_indices", dtype=np.int32)
```
Here you actually never use `spatial_scale`, it is just hardcoded. Please update it.

**Pull request merged**